pkg/httputils/httputils.go

```diff
@@ -25,3 +28,31 @@ func NewHTTPRequestError(msg string, res *http.Response) error {
 		Code:    res.StatusCode,
 	}
 }
+
+type ServerHeader struct {
+	App string // docker
+	Ver string // 1.8.0-dev
+	OS  string // windows or linux
+}
+
+// parseServerHeader extracts pieces from am HTTP server header
+// which is in the format "docker/version (os)" eg docker/1.8.0-dev (windows)
+func ParseServerHeader(hdr string) (*ServerHeader, error) {
+	re := regexp.MustCompile(`.*\((.+)\).*$`)
+	r := &ServerHeader{}
+	if matches := re.FindStringSubmatch(hdr); matches != nil {
+		r.OS = matches[1]
+		parts := strings.Split(hdr, "/")
+		if len(parts) != 2 {
+			return nil, errors.New("Bad header: '/' missing")
+		}
+		r.App = parts[0]
+		v := strings.Split(parts[1], " ")
```

Comment:
It's handling the case of something like docker/1.8.   0  -dev (windows), not just two spaces :)